libc-bottom-half/headers/public/wasi/api.h

```diff
@@ -1625,458 +1589,296 @@ _Static_assert(_Alignof(__wasi_prestat_u_t) == 4, "witx calculated union align")
  */
 __wasi_errno_t __wasi_args_get(
     uint8_t * * argv,
-
     uint8_t * argv_buf
-) __attribute__((
-    __import_module__("wasi_snapshot_preview1"),
-    __import_name__("args_get"),
-    __warn_unused_result__
-));
-
+) __attribute__((__warn_unused_result__));
 /**
  * Return command-line argument data sizes.
+ * @return
+ * Returns the number of arguments and the size of the argument string
+ * data, or an error.
  */
 __wasi_errno_t __wasi_args_sizes_get(
-    /**
-     * The number of arguments.
-     */
-    __wasi_size_t *argc,
-    /**
-     * The size of the argument string data.
-     */
-    __wasi_size_t *argv_buf_size
-) __attribute__((
-    __import_module__("wasi_snapshot_preview1"),
-    __import_name__("args_sizes_get"),
-    __warn_unused_result__
-));
-
+    __wasi_size_t *retptr0,
+    __wasi_size_t *retptr1
```

Comment:
Looks like we lost some information here about what these pointers mean.

Reply:
Yeah unfortunately the documentation got shuffled a bit since each individual parameter is no longer documented in the source file, although I tried to make sure that the documentation was at least preserved somewhat in the `@return` section for the function

Reply:
Any why we can keep the argc and argv_buf_size names some how?

Can these be "retptr_argc" and "retptr_argv_buf_size" perhaps?    retptr0 and retptr1 are not very useful names.

Reply:
Unfortuantely this is mostly due to a change in the `*.witx` file itself -- https://github.com/WebAssembly/WASI/commit/ef8c1a53feb2dfb763d4ea5c7d9e0a0126b45579#diff-2f2ab5a9d79cd5c57a981bf769e772aa60405857803bcdbde3c3e8179315abb8L24-R26

Where previously there were separate `result` variables it's now modeled more closely with the intended interface type, which is a `Result<(usize, usize), Errno>`. This means that there's no way really to auto-generate the names as before since from the source it's just "return pointer for first tuple element" and the second one.

Reply:
There's a trade-off here where we could get names for these values back by returning a record rather than a tuple, but that would break the ABI today (records always turn into a C struct, which gets passed by one pointer rather than multiple). Its a little but unfortunate but I think losing those names is OK for now.